src/space.cpp

```diff
@@ -1027,11 +1027,27 @@ static argval_t do_space(chunk_t *first, chunk_t *second, int &min_sp)
 
    if (chunk_is_token(first, CT_SPAREN_CLOSE))
    {
-      if (  chunk_is_token(second, CT_BRACE_OPEN)
-         && (cpd.settings[UO_sp_sparen_brace].a != AV_IGNORE))
+      if (chunk_is_token(second, CT_BRACE_OPEN))
       {
-         log_rule("sp_sparen_brace");
-         return(cpd.settings[UO_sp_sparen_brace].a);
+         chunk_t *prev = chunk_get_prev_type(first, CT_CATCH, first->level);
```

Comment:
Ick... not entirely your fault, but this is inadequately sophisticated. Consider:
```c++
try { ... }
catch (int) foo();
while (true) { ... }
```

This is affecting the space before the `{` of the `while`. (I suspect it may do that with or without braces around the `catch` block.) You need to instead find the `SPAREN_OPEN` that matches the `SPAREN_CLOSE` (I'd hope we have a sane way of doing that) and check if the preceding token is `CATCH`.

Reply:
What language are you using? This seems like a contrived example since it is not valid C++, Objective-C++, or Java syntax. For these languages the syntax is `catch ( ... ) compound-statement` where `compound-statement` is defined to be `brace-enclosed sequence of statements`.

Reply:
C++. Okay, so I guess for some inexplicable reason, C++ requires braces for a `catch`, even though it almost never requires them otherwise. However, as noted, the issue exists whether or not braces are present. The problem is that `chunk_get_prev_type` tries *really hard* to find the token you requested. AFAICT, the only thing that will stop it is running out of tokens, which means it could match a `catch` in an entirely different *function*... as long as the brace level is the same.

(Maybe it won't span lines, but even so, I could still create a pathological case with two functions on the same line. Granted, I'd probably keel-haul any coder that asked to integrate such code, but... [BLIWYA](https://en.wikipedia.org/wiki/Robustness_principle) :slightly_smiling_face:.)

Reply:
> [`chunk_get_prev_type`] could match a catch in an entirely different *function*...

So... yeah. I tried that, just for fun. And, it does :scream:. Even across multiple lines.